src/sync/sync-utils.ts

```diff
@@ -65,25 +51,53 @@ export const findOrStartSync = async (
 	}, 0, logger);
 };
 
-type SubscriptionUpdateTasks = {
-	totalNumberOfRepos?: number | null;
-	repositoryCursor?: string | null;
-	repositoryStatus?: string | null;
-}
+const getCommitsFromDates = async (jiraHost: string, commitsFromDate: Date | undefined) => {
+	const mainCommitsFromDate = await getCommitSinceDate(jiraHost, NumberFlags.SYNC_MAIN_COMMIT_TIME_LIMIT, commitsFromDate?.toISOString());
+	const branchCommitsFromDate = await getCommitSinceDate(jiraHost, NumberFlags.SYNC_BRANCH_COMMIT_TIME_LIMIT, commitsFromDate?.toISOString());
+	return { mainCommitsFromDate, branchCommitsFromDate };
+};
 
-const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncType, targetTasks?: TaskType[]): Promise<void> => {
-	if (!targetTasks?.length) {
-		return;
+const resetRepoTaskStatusAndCursor = async (subscription: Subscription, syncType: SyncType, targetTasks?: TaskType[]) => {
```

Comment:
As it also updates the subscription, would it make sense to have some more generic name, e.g. "resetSyncStatus" ?

Reply:
Make sense! I was only intended to update repo sync but changed mind later and forgot to update function name.